File: desktop_app/src/main/java/com/model/Producto.java
package com.model;

import java.util.Date;

public class Producto {
    private String codigoBarra;
    private String descripcion;
    private float precioCompra;
    private float precioVenta;
    private int stock;
    private String imgUri;
    private Date fechaCreacion;
    private Date fechaModificacion;
    private Proveedor proveedor;
    private Marca marca;
    private Rubro rubro;

    public Producto() {}

    public Producto(
            String codigoBarra,
            String descripcion,
            float precioCompra,
            float precioVenta,
            int stock,
            String imgUri,
            Date fechaCreacion,
            Date fechaModificacion,
            Proveedor proveedor,
            Marca marca,
            Rubro rubro
    ) {
        this.codigoBarra = codigoBarra;
        this.descripcion = descripcion;
        this.precioCompra = precioCompra;
        this.precioVenta = precioVenta;
        this.stock = stock;
        this.imgUri = imgUri;
        this.fechaCreacion = fechaCreacion;
        this.fechaModificacion = fechaModificacion;
        this.proveedor = proveedor;
        this.marca = marca;
        this.rubro = rubro;
    }

    public String getCodigoBarra() {
        return codigoBarra;
    }

    public void setCodigoBarra(String codigoBarra) {
        this.codigoBarra = codigoBarra;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public float getPrecioVenta() {
        return precioVenta;
    }

    public void setPrecioVenta(float precioVenta) {
        this.precioVenta = precioVenta;
    }

    public float getPrecioCompra() {
        return precioCompra;
    }

    public void setPrecioCompra(float precioCompra) {
        this.precioCompra = precioCompra;
    }

    public int getStock() {
        return stock;
    }

    public void setStock(int stock) {
        this.stock = stock;
    }

    public String getImgUri() {
        return imgUri;
    }

    public void setImgUri(String imgUri) {
        this.imgUri = imgUri;
    }

    public Date getFechaCreacion() {
        return fechaCreacion;
    }

    public void setFechaCreacion(Date fechaCreacion) {
        this.fechaCreacion = fechaCreacion;
    }

    public Date getFechaModificacion() {
        return fechaModificacion;
    }

    public void setFechaModificacion(Date fechaModificacion) {
        this.fechaModificacion = fechaModificacion;
    }

    public Proveedor getProveedor() {
        return proveedor;
    }

    public void setProveedor(Proveedor proveedor) {
        this.proveedor = proveedor;
    }

    public Marca getMarca() {
        return marca;
    }

    public void setMarca(Marca marca) {
        this.marca = marca;
    }

    public Rubro getRubro() {
        return rubro;
    }

    public void setRubro(Rubro rubro) {
        this.rubro = rubro;
    }
}

File: api_rest/src/Producto/domain/Producto.ts
import type { ProductoCodigoBarra } from "./ProductoCodigoBarra";
import type { ProductoDescripcion } from "./ProductoDescripcion";
import type { ProductoFechaCreacion } from "./ProductoFechaCreacion";
import type { ProductoImgUri } from "./ProductoImgUri";
import type { ProductoPrecioCompra } from "./ProductoPrecioCompra";
import type { ProductoPrecioVenta } from "./ProductoPrecioVenta";
import type { ProductoStock } from "./ProductoStock";
import type { ProductoFechaModificacion } from "./ProductoFechaModificacion";
import type { ProveedorId } from "../../Proveedor/domain/ProveedorId";
import type { MarcaId } from "../../Marca/domain/MarcaId";
import type { RubroId } from "../../Rubro/domain/RubroId";

export class Producto {
    codigoBarra: ProductoCodigoBarra;
    descripcion: ProductoDescripcion;
    precioCompra: ProductoPrecioCompra;
    precioVenta: ProductoPrecioVenta;
    stock: ProductoStock;
    imgUri: ProductoImgUri;
    fechaCreacion: ProductoFechaCreacion;
    fechaModificacion: ProductoFechaModificacion;
    proveedorId: ProveedorId;
    marcaId: MarcaId;
    rubroId: RubroId;

    constructor(
        codigoBarra: ProductoCodigoBarra,
        descripcion: ProductoDescripcion,
        precioCompra: ProductoPrecioCompra,
        precioVenta: ProductoPrecioVenta,
        stock: ProductoStock,
        imgUri: ProductoImgUri,
        fechaCreacion: ProductoFechaCreacion,
        fechaModificacion: ProductoFechaModificacion,
        proveedorId: ProveedorId,
        marcaId: MarcaId,
        rubroId: RubroId,
    ) {
        this.codigoBarra = codigoBarra;
        this.descripcion = descripcion;
        this.precioCompra = precioCompra;
        this.precioVenta = precioVenta;
        this.stock = stock;
        this.imgUri = imgUri;
        this.fechaCreacion = fechaCreacion;
        this.fechaModificacion = fechaModificacion;
        this.proveedorId = proveedorId;
        this.marcaId = marcaId;
        this.rubroId = rubroId;
    }

    toJSON() {
        return {
            codigoBarra: this.codigoBarra.value,
            descripcion: this.descripcion.value,
            precioCompra: this.precioCompra.value,
            precioVenta: this.precioVenta.value,
            stock: this.stock.value,
            imgUri: this.imgUri.value,
            fechaCreacion: this.fechaCreacion.value,
            fechaModificacion: this.fechaModificacion.value,
            proveedorId: this.proveedorId.value,
            marcaId: this.marcaId.value,
            rubroId: this.rubroId.value,
        };
    }
}